tensorly/decomposition/tests/test_parafac2.py

```diff
@@ -228,25 +229,26 @@ def test_parafac2_nn():
 
 
 def test_parafac2_slice_and_tensor_input():
+    rng = tl.check_random_state(1234)
     rank = 3
 
     random_parafac2_tensor = random_parafac2(
-        shapes=[(15, 30) for _ in range(25)], rank=rank, random_state=1234
+        shapes=[(15, 30) for _ in range(25)], rank=rank, random_state=rng
     )
     tensor = parafac2_to_tensor(random_parafac2_tensor)
     slices = parafac2_to_slices(random_parafac2_tensor)
 
     slice_rec = parafac2(
-        slices, rank, random_state=1234, normalize_factors=False, n_iter_max=2
```

Comment:
Because the decomposition was being set to the same seed as the random tensor generation, the starting point was exactly at the solution. This was causing the error to be very slightly negative (just due to roundoff) and causing a warning about a negative sqrt in the code. These changes make the test a little better, in principle, because it is not "cheating" by starting at the exact same random point. It also resolves the warning.

Reply:
Great catch, this was a really problematic issue, basically making the test useless!

Reply:
We may check other tests to see if the same issue happens elsewhere, I recall having a similar issue with admm on a local branch